src/Native/Unix/CMakeLists.txt

```diff
@@ -90,7 +90,9 @@ if (UPPERCASE_CMAKE_BUILD_TYPE STREQUAL DEBUG)
       set(CMAKE_SHARED_LINKER_FLAGS_DEBUG "${CMAKE_SHARED_LINKER_FLAGS_DEBUG} ${CLR_SANITIZE_LINK_FLAGS} -Wl,--gc-sections")
     endif ()
 elseif (UPPERCASE_CMAKE_BUILD_TYPE STREQUAL RELEASE)
-    if (CMAKE_SYSTEM_PROCESSOR STREQUAL armv7l AND DEFINED ENV{CROSSCOMPILE})
+    # Use O1 option when the clang version is smaller than 3.9
+    # Otherwise use O3 option in release build
+    if (CMAKE_SYSTEM_PROCESSOR STREQUAL armv7l AND DEFINED ENV{CROSSCOMPILE} AND CMAKE_CXX_COMPILER_VERSION VERSION_LESS 3.9)
```

Comment:
@jyoungyun What do you think of making it simpler - for Arm cross builds, we should expect to use clang 3.9 and error out if not found. This allows us to specify -O3 unconditionally. Also, this should be set independent of the build configuration.

If you agree, can you please also update https://github.com/dotnet/core-setup/commit/7a97793a4c0d3bdc295501ada4b17b3884273946#diff-84bd659d6cb7857c6f9d4f557e88d950R196 to conditionalize using clang 3.9 only for Arm cross builds?